tests/test_collection_info.py

```diff
@@ -0,0 +1,302 @@
+# (c) 2012-2019, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import pytest
+
+from galaxy_importer.schema import CollectionInfo
+
+
+@pytest.fixture
+def collection_info():
+    metadata = {
+        'namespace': 'acme',
+        'name': 'jenkins',
+        'version': '3.5.0',
+        'license': ['MIT'],
+        'readme': 'README.rst',
+        'authors': ['Bob Smith <b.smith@acme.com>'],
+        'tags': ['testcases']
+    }
+    return metadata
+
+
+def test_collection_info(collection_info):
+    res = CollectionInfo(**collection_info)
+    assert type(res) == CollectionInfo
+
+
+def test_readme_req(collection_info):
+    collection_info['readme'] = ''
+    with pytest.raises(ValueError) as exc:
+        CollectionInfo(**collection_info)
+        assert "'readme' is required by galaxy" in str(exc)
```

Comment:
This actually needs to be outside the `with` block. When `CollectionInfo(...)` raises, it exits the block. 

Also, `exc` here is not the exception object but an `ExceptionInfo` wrapper Pytest uses. You can get the actual exception object you want to stringify with `exc.value`.

Reply:
thanks, i forgot to add the `match=` pattern here